cli/running/lua/cat.lua

```diff
@@ -0,0 +1,265 @@
+    -- body (new), and the one with empty body (old).
+    if record.HEADER.type == 'NOP' or record.BODY == nil or
+    record.BODY.space_id == nil then
+        return
+    end
+    io.stdout:write(('box.space[%d]'):format(record.BODY.space_id))
+    local op = record.HEADER.type:lower()
+    io.stdout:write((':%s('):format(op))
+    if op == 'insert' or op == 'replace' then
+        write_lua_table(record.BODY.tuple)
+    elseif op == 'delete' then
+        write_lua_table(record.BODY.key)
+    elseif op == 'update' then
+        write_lua_table(record.BODY.key)
+        io.stdout:write(', ')
+        write_lua_table(record.BODY.tuple)
+    elseif op == 'upsert' then
+        write_lua_table(record.BODY.tuple)
+        io.stdout:write(', ')
+        write_lua_table(record.BODY.operations)
+    end
+    io.stdout:write(')\n')
+end
+
+local cat_formats = setmetatable({
+    yaml = cat_yaml_cb,
+    json = cat_json_cb,
+    lua  = cat_lua_cb,
+}, {
+    __index = function(self, cmd)
+        log.error("Internal error: unknown formatter '%s'", cmd)
+        os.exit(1)
+    end
+})
+
+local function find_in_list(id, list)
+    if type(list) == 'number' then
+        return id == list
+    end
+    for _, val in ipairs(list) do
+        if val == id then
+            return true
+        end
+    end
+    return false
+end
+
+local function filter_xlog(gen, param, state, opts, cb)
+    local from, to, spaces = opts.from, opts.to, opts.space
+    local show_system, replicas = opts['show-system'], opts.replica
+
+    for lsn, record in gen, param, state do
+        local sid = record.BODY and record.BODY.space_id
+        local rid = record.HEADER.replica_id
+        if replicas and #replicas == 1 and replicas[1] == rid and lsn >= to then
+            -- Stop, as we've finished reading tuple with lsn == to
+            -- and the next lsn's will be bigger.
+            break
+        elseif (lsn < from) or (lsn >= to) or
+        (not spaces and sid and sid < 512 and not show_system) or
+        (spaces and (sid == nil or not find_in_list(sid, spaces))) or
+        (replicas and not find_in_list(rid, replicas)) then
+            -- Pass this tuple, luacheck: ignore.
+        else
+            cb(record)
+        end
+    end
+end
+
+local function cat()
+    local opts = keyword_arguments
+    local cat_format = opts.format
+    local format_cb = cat_formats[cat_format]
+    local is_printed = false
+    for _, file in ipairs(positional_arguments) do
+        print(string.format("• Result of cat: the file '%s' is processed below •", file))
+        local gen, param, state = xlog.pairs(file)
+        filter_xlog(gen, param, state, opts, function(record)
+            is_printed = true
+            format_cb(record)
+            io.stdout:flush()
+        end)
+        if opts.format == 'yaml' and is_printed then
+            is_printed = false
+            print('...\n')
+        end
+    end
+end
+
+local function populate_arguments()
```

Comment:
Hmm, AFAIU you pass all arguments through env variables. What do you need this function for?

Reply:
Simplified, `populate_arguments()` has been removed

Reply:
> Ouch, did I understand right that you add all the arguments to `arg` and after parse them? What for such overkill?

Yep, it was overkill